chef/cookbooks/ohai/files/default/plugins/linux/s390x.rb

```diff
@@ -0,0 +1,40 @@
+provides "s390x"
+
+if File.exist? "/proc/sysinfo"
+  retval = Mash.new.tap do |result|
+    result[:system] = {}
+
+    result[:system][:manufacturer] = "PR/SM"
+
+    File.open("/proc/sysinfo") do |p|
```

Comment:
Nitpick: even though this temporary variable is only used once, I'd still prefer to avoid cryptic variable names, especially ones which are the same as a commonly used method in `Kernel`.  Maybe just `file` ?